lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,74 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    # collection
+
+    # Version 1, If using variables:
+    # new_enum = collection.map { |item| item.class}
+    # total = new_enum.count(Fixnum) + new_enum.count(Float)
+
+    # Version 2, .map vs .select
+    # new_arr2 = collection.map { |item| item.is_a? (Numeric)} #=> [true, true, false, true, true, false, false, true, true, false, true]
+    # new_arr3 = collection.select { |item| item.is_a? (Numeric)} #=> [1, 2, 3, 4, 5, 6, 3.14]
+
+    collection.select { |item| item.is_a? (Numeric)}.count
   end
 
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
   def all_words_longer_than_length?(min_length)
+    # ea_word_len = collection.all? { |item| item.length > min_length}
+    collection.all? { |item| item.length > min_length}
   end
 
 
   # For the next 3 methods refer to: collect or map
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-any-3F
   def contains_a_word_longer_than?(min_length)
+    collection.map { |item| item.length > min_length}.include? true
   end
 
   def capitalize_words
+    # collection.map { |item| item.capitalize}
+
+    # the symbol to proc way
+    collection.map(&:capitalize)
   end
 
   def square_numbers
+    collection.map { |item| item * item}
   end
 
   # Use symbol to proc notation on this one!
   def reverse_words
+    collection.map(&:reverse)
   end
 
   # Refer to collect_concat or flat_map for these next two
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-flat_map
   # To get the order right for the tests, you'll need to use the absolute value method
   def positive_and_negative_numbers
+    collection.map(&:abs).flat_map {|item| [item, -item]}
+
   end
 
   # Refer to detect or find
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-detect
   def find_first_awesome_person
+    collection.find { |hash| hash[:awesome] == true}
```

Comment:
It's more accurate to name your local variable by what it represents, so`|person|` instead of `|hash|`